prow/apis/prowjobs/v1/types_test.go

```diff
@@ -0,0 +1,374 @@
+					Bucket:       "bucket",
+					PathPrefix:   "prefix",
+					PathStrategy: PathStrategyLegacy,
+					DefaultOrg:   "org",
+					DefaultRepo:  "repo",
+				},
+				GCSCredentialsSecret: "secretName",
+				SSHKeySecrets:        []string{"first", "second"},
+				SSHHostFingerprints:  []string{"primero", "segundo"},
+				SkipCloning:          &truth,
+			},
+		},
+		{
+			name: "gcs configuration partially provided",
+			provided: &DecorationConfig{
+				GCSConfiguration: &GCSConfiguration{
+					Bucket: "bucket-1",
+				},
+			},
+			expected: &DecorationConfig{
+				Timeout:     1 * time.Minute,
+				GracePeriod: 10 * time.Second,
+				UtilityImages: &UtilityImages{
+					CloneRefs:  "clonerefs",
+					InitUpload: "iniupload",
+					Entrypoint: "entrypoint",
+					Sidecar:    "sidecar",
+				},
+				GCSConfiguration: &GCSConfiguration{
+					Bucket:       "bucket-1",
+					PathPrefix:   "prefix",
+					PathStrategy: PathStrategyLegacy,
+					DefaultOrg:   "org",
+					DefaultRepo:  "repo",
+				},
+				GCSCredentialsSecret: "secretName",
+				SSHKeySecrets:        []string{"first", "second"},
+				SSHHostFingerprints:  []string{"primero", "segundo"},
+				SkipCloning:          &truth,
+			},
+		},
+		{
+			name: "skip_cloning provided",
+			provided: &DecorationConfig{
+				SkipCloning: &lies,
+			},
+			expected: &DecorationConfig{
+				Timeout:     1 * time.Minute,
+				GracePeriod: 10 * time.Second,
+				UtilityImages: &UtilityImages{
+					CloneRefs:  "clonerefs",
+					InitUpload: "iniupload",
+					Entrypoint: "entrypoint",
+					Sidecar:    "sidecar",
+				},
+				GCSConfiguration: &GCSConfiguration{
+					Bucket:       "bucket",
+					PathPrefix:   "prefix",
+					PathStrategy: PathStrategyLegacy,
+					DefaultOrg:   "org",
+					DefaultRepo:  "repo",
+				},
+				GCSCredentialsSecret: "secretName",
+				SSHKeySecrets:        []string{"first", "second"},
+				SSHHostFingerprints:  []string{"primero", "segundo"},
+				SkipCloning:          &lies,
+			},
+		},
+		{
+			name: "ssh host fingerprints provided",
+			provided: &DecorationConfig{
+				SSHHostFingerprints: []string{"unique", "print"},
+			},
+			expected: &DecorationConfig{
+				Timeout:     1 * time.Minute,
+				GracePeriod: 10 * time.Second,
+				UtilityImages: &UtilityImages{
+					CloneRefs:  "clonerefs",
+					InitUpload: "iniupload",
+					Entrypoint: "entrypoint",
+					Sidecar:    "sidecar",
+				},
+				GCSConfiguration: &GCSConfiguration{
+					Bucket:       "bucket",
+					PathPrefix:   "prefix",
+					PathStrategy: PathStrategyLegacy,
+					DefaultOrg:   "org",
+					DefaultRepo:  "repo",
+				},
+				GCSCredentialsSecret: "secretName",
+				SSHKeySecrets:        []string{"first", "second"},
+				SSHHostFingerprints:  []string{"unique", "print"},
+				SkipCloning:          &truth,
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		if actual, expected := testCase.provided.ApplyDefault(defaults), testCase.expected; !reflect.DeepEqual(actual, expected) {
```

Comment:
consider making test cases not have to worry about overriding the defaults

```golang
caseDefault := *defaults
if a, e := testCase.provided.ApplyDefault(&caseDefault)...
```

Reply:
Done.